File: problemList/slidingWindow/solution/LC209.java
package problemList.slidingWindow.solution;

public class LC209 {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length, sum = 0, left = 0, ret = Integer.MAX_VALUE;
        for(int i = 0;i < n;i++){
            sum += nums[i];
            while(left <= i && sum >= target){
                ret = Math.min(ret, i - left + 1);
                sum -= nums[left];
                left++;
            }
        }
        return ret == Integer.MAX_VALUE ? 0 : ret;
    }
}

File: SolutionsWithCorCpp/problem_list_0x3f/sliding_window/solution/LC209.cpp
#include <algorithm>
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

int minSubArrayLen(int target, vector<int> &nums) {
	int n = nums.size(), r = 0, ret = INT_MAX;
	long long curSum = 0;
	for (int i = 0; r < n; r++) {
		curSum += nums[r];
		while (curSum >= target && i < n) {
			if (i <= r) ret = min(ret, r - i + 1);
			curSum -= nums[i];
			i++;
		}
	}
	return ret == INT_MAX ? 0 : ret;
}